db/memtable.cc

```diff
@@ -1150,11 +1176,21 @@ static bool SaveValue(void* arg, const char* entry) {
       case kTypeRangeDeletion: {
         if (*(s->merge_in_progress)) {
           if (s->value || s->columns) {
+            std::string result;
             *(s->status) = MergeHelper::TimedFullMerge(
                 merge_operator, s->key->user_key(), nullptr,
-                merge_context->GetOperands(), s->value, s->columns, s->logger,
-                s->statistics, s->clock, /* result_operand */ nullptr,
+                merge_context->GetOperands(), &result, s->logger, s->statistics,
+                s->clock, /* result_operand */ nullptr,
                 /* update_num_ops_stats */ true);
+
+            if (s->status->ok()) {
+              if (s->value) {
```

Comment:
IIUC, in case of merge, for general flow value is a valid pointer and in case of wide columns value is nullptr so result is stored in the columns.

Reply:
Right. These code paths are shared by three APIs: for `GetMergeOperands`, `do_merge` is false; for `Get`, `do_merge` is true and `value` is non-null (and `columns` is null); and for `GetEntity`, `do_merge` is true and `columns` non-null (and `value` is null).